src/test/java/com/amazonaws/services/kinesis/clientlibrary/config/KinesisClientLibConfiguratorTest.java

```diff
@@ -189,6 +204,49 @@ public void testWithInitialPositionInStreamVariables() {
                 }, '\n'));
 
         assertEquals(config.getInitialPositionInStream(), InitialPositionInStream.TRIM_HORIZON);
+    }    
+
+    @Test
+    public void testWithTimestampAtInitialPositionInStreamVariables() {
+        KinesisClientLibConfiguration config =
+                getConfiguration(StringUtils.join(new String[] {
+                        "streamName = a",
+                        "applicationName = b",
+                        "AWSCredentialsProvider = ABCD," + credentialName1,
+                        "timestampAtInitialPositionInStream = 1527267472"
+                }, '\n'));
+
+        assertEquals(config.getInitialPositionInStream(), InitialPositionInStream.AT_TIMESTAMP);
+        assertEquals(config.getTimestampAtInitialPositionInStream(), 
+                new Date(1527267472 * 1000L));
+    }
+
+    @Test
+    public void testWithEmptyTimestampAtInitialPositionInStreamVariables() {
+        KinesisClientLibConfiguration config =
+                getConfiguration(StringUtils.join(new String[] {
+                        "streamName = a",
+                        "applicationName = b",
+                        "AWSCredentialsProvider = ABCD," + credentialName1,
+                        "timestampAtInitialPositionInStream = "
+                }, '\n'));
+
+        assertEquals(config.getInitialPositionInStream(), InitialPositionInStream.LATEST);
+        assertEquals(config.getTimestampAtInitialPositionInStream(), null);
+    }
+
+    @Test
+    public void testWithNonNumericTimestampAtInitialPositionInStreamVariables() {
+        KinesisClientLibConfiguration config =
+                getConfiguration(StringUtils.join(new String[] {
+                        "streamName = a",
+                        "applicationName = b",
+                        "AWSCredentialsProvider = ABCD," + credentialName1,
+                        "timestampAtInitialPositionInStream = 123abc"
```

Comment:
Calling this out for myself, and future readers: 

Looking at the code this will at least log a warning.  That isn't part of your code, but I don't know if that is a good outcome.  If the InitialPositionInStream is set to AT_TIMESTAMP then the KCL will be unable to startup if the lease table doesn't exist or isn't populated.  Otherwise the KCL should be able to startup, but that could be really confusing.